src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -108,29 +108,33 @@ def __init__(
 
 
 class GroupingSets:
-    """Creates a :class:`GroupingSets` object from a list of column/expression sets that you pass
-    to :meth:`DataFrame.group_by_grouping_sets`. See :meth:`DataFrame.group_by_grouping_sets` for
-    examples of how to use this class with a :class:`DataFrame`. See
-    `GROUP BY GROUPING SETS <https://docs.snowflake.com/en/sql-reference/constructs/group-by-grouping-sets.html>`_
-    for its counterpart in SQL (several examples are shown below).
-
-    =============================================================  ==================================
-    Python interface                                               SQL interface
-    =============================================================  ==================================
-    ``GroupingSets([col("a")], [col("b")])``                       ``GROUPING SETS ((a), (b))``
-    ``GroupingSets([col("a") , col("b")], [col("c"), col("d")])``  ``GROUPING SETS ((a, b), (c, d))``
-    ``GroupingSets([col("a"), col("b")])``                         ``GROUPING SETS ((a, b))``
-    ``GroupingSets(col("a"), col("b"))``                           ``GROUPING SETS ((a, b))``
-    =============================================================  ==================================
-    """
-
-    def __init__(self, *sets: Union[Column, List[Column]]) -> None:
-        self._ast = with_src_position(proto.SpGroupingSets())
-        set_list, self._ast.sets.variadic = parse_positional_args_to_list_variadic(
-            *sets
-        )
-        for s in set_list:
-            build_expr_from_python_val(self._ast.sets.args.add(), s)
+    @publicapi
+    def __init__(
+        self, *sets: Union[Column, List[Column]], _emit_ast: bool = True
+    ) -> None:
```

Comment:
found it weird that _init_ didn't have a docstring, moved doc here.

Reply:
have you checked the effect on docstring this generates? I see a similar pattern in other files like `DataFrame`. Can you please verify the docstrings are unaffected